modules/javafx.controls/src/main/java/javafx/scene/control/skin/TreeTableCellSkin.java

```diff
@@ -104,8 +104,42 @@ public TreeTableCellSkin(TreeTableCell<S,T> control) {
     }
 
     /** {@inheritDoc} */
-    @Override double leftLabelPadding() {
-        double leftPadding = super.leftLabelPadding();
+    @Override
+    protected double computeMinWidth(double height, double topInset, double rightInset, double bottomInset,
+            double leftInset) {
+        return super.computeMinWidth(height, topInset, rightInset, bottomInset, leftInset) + calculateIndentation();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    protected void layoutChildren(double x, double y, double w, double h) {
+        double indentation = calculateIndentation();
+        x += indentation;
+        w -= indentation;
+        super.layoutChildren(x, y, w, h);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    protected double computePrefWidth(double height, double topInset, double rightInset, double bottomInset,
+            double leftInset) {
+        if (isDeferToParentForPrefWidth) {
+            return super.computePrefWidth(height, topInset, rightInset, bottomInset, leftInset) + calculateIndentation();
```

Comment:
If we are deferring to Parent for the preferred width, then we need not add indentation. 
I think, we should interchange line 127 and line 130.

Reply:
No that's actually correct as it is. 

That is because when a cell is deferred from the parent we want to know the width we need without being truncated. For now that is only set when the auto size of the table is triggered (see also: **TableColumnHeader#resizeColumnToFitContent**). 

And if not deferred, the width of the cell is in sync with the column (you can also see it in the **super.computePrefWidth**)

Reply:
OK. Thanks for the explanation.